### PR TITLE
fix: uid availability check

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

Fixes about the UID field availibility check:
- the check should not run if you haven't changed the value of the uid field
- the availability cache should be cleared when you save or update a content type

Also fixes the initialValue from the useField hook which was broken

### How to test it?

- Create a content type with a UID field
- Save an entry with a value in the UID field
- You should no longer see "available" or "unavailable" when nothing was changed
- Create a new entry in the same content type
- If you enter the same UID as the first entry, it should say unavailable. Anything else should be available

### Related issue(s)/PR(s)

fixes [#20598](https://github.com/strapi/strapi/issues/20598)

### CODE REVIEW

**Review by madhurisandbhor — COMMENTED** [review]
https://github.com/user-attachments/assets/7f158c9e-b122-404a-b8f0-39d1552dc839

Saving empty value doesn't clear unavailability check

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/tests/UID.test.tsx [hunk 1]
```diff
@@ -92,61 +102,56 @@ describe('UIDInput', () => {
...
-      timeout: 10000,
+    await waitFor(() => expect(screen.queryByTestId('loading-wrapper')).not.toBeInTheDocument());
+    expect(await screen.findByText(/^Available$/)).toBeInTheDocument();
+
+    // Change the value to make it unavailable
+    await user.type(input, 'unavailable');
+    act(() => {
+      jest.advanceTimersByTime(4000);
     });
+    expect(await screen.findByText(/^Unavailable$/)).toBeInTheDocument();
```
Comment (madhurisandbhor) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/tests/UID.test.tsx:141]: Could you please use different uid than `available` and `unavailable`, I am not sure why your unavailability check is on here?

Comment (remidej) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/tests/UID.test.tsx]: I can see how the uid is confusing so I updated it.

Regarding the check itself, I'm testing the general availability check behavior, so it makes sense to me to check both outcomes, the positive and negative results

**Review by madhurisandbhor — COMMENTED** [review]
I am still unable to save empty uid value(if its not required)

https://github.com/user-attachments/assets/f5a9e000-c738-4821-bba1-0e82507b8061

**Comment by remidej** [comment]
I'm opening a separate PR for the unique check (comes from the backend and also affects normal strings, not just UIDs)

**Pull request merged**